src/devices/sunricher.ts

```diff
@@ -361,7 +362,159 @@ async function syncTime(endpoint: Zh.Endpoint) {
+        {
+            key: [
+                'pairing_duration',
+                'light_pwm_frequency',
+                'manufacture_id',
+                'brightness_curve',
+                'start_up_on_off',
+                'motion_sensor_light_duration',
+                'motion_sensor_light_sensitivity',
+                'motion_sensor_working_mode',
+                'motion_sensor_sensing_distance',
+                'motion_sensor_microwave_switch',
+                'motion_sensor_onoff_broadcast',
+                'motion_sensor_light_state',
+                'motion_sensor_in_pwm_brightness',
+                'motion_sensor_in_pwm_output',
+                'motion_sensor_leave_pwm_output',
+                'motion_sensor_leave_delay',
+                'motion_sensor_pwm_output_after_delay',
+                'linear_error_ratio_coefficient_of_lux_measurement',
+                'fixed_deviation_of_lux_measurement',
+            ],
+            convertSet: async (entity, key, value, meta) => {
+                const lookup = {
+                    pairing_duration: {attribute: 0x9000, type: 0x20},
+                    light_pwm_frequency: {attribute: 0x9001, type: 0x21},
+                    manufacture_id: {attribute: 0x9003, type: 0x21},
+                    brightness_curve: {attribute: 0x8806, type: 0x20},
+                    start_up_on_off: {attribute: 0x4003, type: 0x30},
+                    motion_sensor_light_duration: {attribute: 0x8902, type: 0x21},
+                    motion_sensor_light_sensitivity: {attribute: 0x8905, type: 0x20},
+                    motion_sensor_working_mode: {attribute: 0x8904, type: 0x20},
+                    motion_sensor_sensing_distance: {attribute: 0x8905, type: 0x20},
+                    motion_sensor_microwave_switch: {attribute: 0x8906, type: 0x20},
+                    motion_sensor_onoff_broadcast: {attribute: 0x8907, type: 0x20},
+                    motion_sensor_light_state: {attribute: 0x890c, type: 0x20},
+                    motion_sensor_in_pwm_brightness: {attribute: 0x8908, type: 0x21},
+                    motion_sensor_in_pwm_output: {attribute: 0x8909, type: 0x20},
+                    motion_sensor_leave_pwm_output: {attribute: 0x890a, type: 0x20},
+                    motion_sensor_leave_delay: {attribute: 0x8901, type: 0x21},
+                    motion_sensor_pwm_output_after_delay: {attribute: 0x890b, type: 0x20},
+                    linear_error_ratio_coefficient_of_lux_measurement: {attribute: 0x890d, type: 0x21},
+                    fixed_deviation_of_lux_measurement: {attribute: 0x890e, type: 0x29},
+                };
+
+                const attribute = lookup[key as keyof typeof lookup];
+                await entity.write('genBasic', {[attribute.attribute]: {value: value, type: attribute.type}}, {manufacturerCode: 0x1224});
+                return {state: {[key]: value}};
+            },
+        },
+    ];
+
+    const exposes: Expose[] = [
+        e
```

Comment:
Could you use modern extend for all of these? Examples: https://github.com/Koenkk/zigbee-herdsman-converters/blob/9bb99f0c972c5a339c846ac966bf2739d591fe42/src/devices/sonoff.ts#L1180

Reply:
DONE